### ISSUE TITLE
Arch Linux man page missing

### ISSUE THREAD

**Issue by alerque** [simonmichael/hledger#1050]
I see from #282 (and the source code of the man page) that there should be one, but `man hledger` turns up nothing on my system. This _may_ be an issue with the Arch Linux [PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/hledger) but I didn't see anything obvious. As I'm not entirely clear on how the Haskell build and install system works I'm not sure what I'm looking for.

**Comment by simonmichael** [comment]
Thanks for the report. This probably needs to be reported to the Arch packagers - the man pages are included in the release tarballs they get from Hackage.

**Comment by alerque** [comment]
I have [reported this upstream](https://bugs.archlinux.org/task/62891).

**Comment by alerque** [comment]
Note the shell completion routines are not properly packaged on Arch. I've [reported it here](https://bugs.archlinux.org/task/63197).

**Comment by felixonmars** [comment]
@alerque @simonmichael Sorry for the very late response. Would it be possible to include the bash completion files in hackage source tarballs? Currently the manpage is included but the bash completion file is not.

**Comment by simonmichael** [comment]
@felixonmars could you open a new issue for this. Maybe @zhelezov or @schoettl could take a look. Related: #1410.

**Comment by schoettl** [comment]
I just looked at it but I coudn't find out where and how the tarball is generated. I searched for `tar.gz` but it's only mentioned in one single comment in the Makefile.

Regarding the missing man page on Arch, I noticed that all man pages in my `/usr/share/man/man1/` are .1.gz files which leads me to the question if the PKGBUILD must call gzip?

Regarding the missing shell completions, @simonmichael should we add just a new section SHELL COMPLETIONS to `Shake.hs`?

Something like this

```hs
      phony "txtmanuals" $ need txtmanuals
      txtmanuals |%> \out -> do  -- hledger/hledger.txt
        let src = manualNameToManpageName $ dropExtension out
        ...
        cmd Shell "tbl" src "| eqn -Tascii | troff -Wall -mandoc -Tascii | grotty -cbuo >" out

```
seems to copy files to the "output" directory `haskell/`. So we could add a similar (but simpler) rule for the shell completions.

**Comment by simonmichael** [comment]
The hackage tarball contents are defined by the .cabal files (and therefore, the package.yaml files).

**Comment by alerque** [comment]
@schoettl No, most man packaging systems handle the compression step on their own, upstream projects can and typically do distribute (or generate) just the `*.1` or whatever man page source files.

**Comment by simonmichael** [comment]
I'm not aware of the missing shell completions issue, but happy to look at a PR for the shakefile or makefile to help automate generating things at release time.  Your choice, I typically use the makefile for easier stuff, shakefile for complex/sensitive stuff.

**Comment by felixonmars** [comment]
The man pages are now available in the Arch packages since 1.20.3. Sorry again for the delay.

**Comment by alerque** [comment]
I think this can be marked as done. Thanks @felixonmars! The completions being in the `zsh-completions` package instead of `hledger` is a bit weird, but that can probably be overhauled when #1410 lands with much improved completion scripts and a proper way to generate them on build.
